packages/js/plugins/ws/README.md

```diff
@@ -0,0 +1,130 @@
+# @polywrap/ws-plugin-js
+
+WebSocket plugin allows Polywrap Client to interact with WebSocket servers.
+
+## interface
+
+``` typescript
+# subset of JS MessageEvent interface
+type Message {
+    data: String!
+    origin: String!
+    lastEventId: String!
+}
+
+# path to WRAP method
+type Callback {
+    uri: String!,
+    method: String!
+}
+
+# optional fields are `Number | null` instead of `Option<i32>`
+type Number {
+    value: Int!
+}
```

Comment:
Optional Int is currently implemented as Option<i32> in wasm and schema/bind.
But to omit optional fields from the function parameters we need a " | null" union type.
So we wrap Int in a reference type and make it optional, "Number | null".

I mentioned this in #944, but now I think we might want to implement Optional Int as "Option<i32> | null" and always use null instead of Option.None<i32>().

Reply:
i.e. `Box<i32> | null` instead of `Option<i32> | null`

Reply:
Exactly, created an issue for that #1132.